gladier_kanzus/tools/dials_stills.py

```diff
@@ -1,37 +1,42 @@
 from gladier import GladierBaseTool, generate_flow_definition
 
-def stills_process(**data):
+def dials_stills(**data):
     import os
     import subprocess
-    from subprocess import PIPE
 
-    
     proc_dir = data['proc_dir']
     data_dir = data['data_dir']
-    input_files = data['input_files']
-    run_num = data['input_files'].split("_")[-2]
-    
-    if 'suffix' in data:
-        phil_name = f"{proc_dir}/process_{run_num}_{data['suffix']}.phil"
-    else:
-        phil_name = f"{proc_dir}/process_{run_num}.phil"
+    run_num = data['run_num']
+    chip_name = data['chip_name']
+    cbf_num = data['cbf_num']
+    batch_size = data['stills_batch_size']
 
-    file_end = data['input_range'].split("..")[-1]
-  
-    if not "timeout" in data:
-        data["timeout"] = 1200
+    phil_name = f"{proc_dir}/process_{run_num}.phil"
 
-    dials_path = data.get('dials_path','')
-    cmd = f'source {dials_path}/dials_env.sh && timeout {data["timeout"]} dials.stills_process {phil_name} {data_dir}/{input_files} > log-{file_end}.txt'
+    cbf_start = cbf_num - batch_size
+    cbf_end = cbf_num - 1
+
+    file_start = f"{chip_name}_{run_num}_{cbf_start}.cbf"
+    file_end = f"{chip_name}_{run_num}_{cbf_end}.cbf"
 
+    input_files = f"{chip_name}_{run_num}_{{{str(cbf_start).zfill(5)}..{str(cbf_end).zfill(5)}}}.cbf"
```

Comment:
Does this use bash expansion to pass the input files?

Reply:
Yes. It is the same bash-expansion as before. I want to change it to create a file instead but did not test it yet.